packages/salesforcedx-vscode-replay-debugger/test/unit/core/logContext.test.ts

```diff
@@ -109,41 +135,135 @@ describe('LogContext', () => {
     parseLogEventStub = sinon
       .stub(LogContext.prototype, 'parseLogEvent')
       .returns(new NoOpState());
+    context.setState(new LogEntryState());
+    context.getFrames().push({} as StackFrame);
 
-    context.updateFrames();
+    context.updateFrames(debugConsoleHandler);
 
     expect(context.getLogLinePosition()).to.equal(1);
+    expect(context.hasState()).to.be.true;
+    expect(context.getFrames()).to.be.empty;
   });
 
   describe('Log event parser', () => {
```

Comment:
@DatGuyJonathan - Is it possible for a user to _deliberately_ change her `System.debug` to produce somethign like

```
13:37:57.194 (291785869)|USER_DEBUG|[46]|DEBUG|End placeholder for setting breakpoint
<newline>
|METHOD_ENTRY|Bogus|...
<newline>
```
Does our log parser get confused by that? 

We cannot provide such deliberate printing, but I'm curious what happens in such cases.

Reply:
Yea, that bogus line will confuse the state. We should chat about what checks are safe & future proof to make this more resilient (e.g. number of fields for different types of events, field types, etc.)